lib/validator/validator.go

```diff
@@ -112,6 +112,10 @@ func init() {
 	registerStructValidator(validateICMPFields, api.ICMPFields{})
 	registerStructValidator(validateRule, api.Rule{})
 	registerStructValidator(validateBackendRule, model.Rule{})
+	registerStructValidator(validateBackendEndpointPort, model.EndpointPort{})
+	registerStructValidator(validateEndpointPort, api.EndpointPort{})
+	registerStructValidator(validateWorkloadEndpoint, model.WorkloadEndpoint{})
```

Comment:
insert "Backend" in the name here, for consistency

Reply:
Also, when do these backend model validators come into play?  (I guess they do, since there are already others above - but don't immediately see the scenario.)

Reply:
Two places: 

- Felix uses them before feeding models into the calculation graph. (So they prevent someone writing rubbish to etcd directly.)
- I think calicoctl applies both front and backend validation (but the errors are cleaner with the former).